### PR TITLE
[glog] Update to 0.5.0

### LINKED ISSUE #17731: [glog] update to 0.5.0
Library name: glog

New version number: https://github.com/google/glog/releases/tag/v0.5.0

Other information that may be useful (release notes, etc...)

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?  
  Updates glog to 0.5.0

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  As before, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run ./vcpkg x-add-version --all and committed the result?  
  Yes

### CODE REVIEW

File: ports/glog/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "glog",
-  "version-string": "0.4.0",
-  "port-version": 4,
+  "version-string": "0.5.0",
```
Comment (JonLiu1993) [inline ports/glog/vcpkg.json:3]:
```suggestion
  "version": "0.5.0",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@chausner ,Thanks for your pr,this is error log:
https://dev.azure.com/vcpkg/public/_build/results?buildId=60211&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by chausner** [comment]
This currently breaks several dependent packages. I also think there is a bug in glog 0.5.0 which I reported at https://github.com/google/glog/issues/715 and tried to workaround with `glog_fix_os_defines.patch` but it appears there are more issues..

**Comment by JonLiu1993** [comment]
@chausner, any progress?

**Comment by chausner** [comment]
> @chausner, any progress?

No, I'll close this PR.

**Pull request closed without merging**